### PR TITLE
feat: adds index exists

### PR DESCRIPTION
**Pull request by chloelbn**
Added a method to check if index exists

| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no
| Related Issue     | Fix https://github.com/algolia/algoliasearch-client-python/issues/434
| Need Doc update   | yes

## Describe your change

Add a method to check if the index exists

### CODE REVIEW

File: algoliasearch/search_index.py [hunk 1]
```diff
@@ -45,6 +45,15 @@ def __init__(self, transporter, config, name):
         self._config = config
         self._name = name
 
+    def exists(self):
```
Comment (nunomaduro) [inline algoliasearch/search_index.py]: Can you add mypy annotations here ?

File: tests/unit/test_search_index.py [hunk 2]
```diff
@@ -29,6 +29,24 @@ def test_app_id_getter(self):
     def test_name_getter(self):
         self.assertEqual(self.index.name, 'index-name')
 
+    def test_index_does_not_exist(self):
```
Comment (nunomaduro) [inline tests/unit/test_search_index.py]: Would be cool to have 2 feature tests for the exist method - in the feature folder. Just like on the CTS. And 1 unit test for testing that when the code is different from 404 you raise the exception.

**Comment by chloelbn** [comment]
@nunomaduro  I approve here as I'm the one who opened the PR. Good job 👍

File: algoliasearch/search_index_async.py [hunk 3]
```diff
@@ -77,6 +77,18 @@ def browse_rules_async(self, request_options=None):
         return RuleIteratorAsync(self._transporter_async, self._name,
                                  request_options)
 
+    def exists_async(self):
+        # type: () -> bool
+
+        try:
+            yield from self.get_settings()
+        except RequestException as e:
```
Comment (chloelbn) [inline algoliasearch/search_index_async.py]: don't forget to import RequestException

Comment (nunomaduro) [reply algoliasearch/search_index_async.py]: Tests are passing, weird.

**Pull request merged**